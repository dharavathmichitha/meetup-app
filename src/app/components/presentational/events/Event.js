import React from 'react';
import PropTypes from 'prop-types';
import Title from '../common/Title';
import Button from '../common/Button';

const Event = ({ event, eventStartTime, eventDuration, cityName }) => {
  const { isFree, name } = event;
  const cityNameAndDuration = `${cityName} - ${`${eventDuration}h`}`;

  return (
    <>
      <div className="event" role="listitem">
        <div className="event__time">{eventStartTime}</div>
        <div className="event__info">
          <Title text={name} isFree={isFree} />
          <div className="event__info__city-name">{cityNameAndDuration}</div>
        </div>
        <Button />
      </div>
    </>
  );
};

Event.propTypes = {
  eventStartTime: PropTypes.string.isRequired,
  eventDuration: PropTypes.number.isRequired,
  cityName: PropTypes.string.isRequired,
  event: PropTypes.shape({
    id: PropTypes.number,
    isFree: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    city: PropTypes.number,
    startDate: PropTypes.string,
    endDate: PropTypes.string,
  }).isRequired,
};

export default Event;
