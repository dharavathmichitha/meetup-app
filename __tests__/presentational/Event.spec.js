import React from 'react';
import renderer from 'react-test-renderer';
import Event from '../../src/app/components/presentational/events/Event';

beforeEach(() => {
  jest.resetModules();
});

describe('Event Component', () => {
  it('should match snapshot', () => {
    const event = {
      id: 0,
      isFree: false,
      name: 'testEvent',
      city: 1,
      startDate: new Date().toDateString(),
      endDate: new Date(1).toDateString(),
    };

    const props = {
      event,
      eventStartTime: '19:00',
      eventDuration: 1,
      cityName: 'Mallorca',
      singUpModalText: 'This is a test modal',
      displayModal: true,
      toggleModalDisplay: () => true,
      eventIsAlreadyRegistered: true,
      signUpHandler: () => true,
      cancelHandler: () => true,
    };

    const component = renderer.create(<Event {...props} />);

    const componentTree = component.toJSON();

    expect(componentTree).toMatchSnapshot();
  });
});
