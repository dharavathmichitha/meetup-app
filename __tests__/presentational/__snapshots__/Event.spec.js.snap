// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component should match snapshot 1`] = `
<div
  className="event"
  role="listitem"
>
  <div
    aria-level="3"
    className="event__time"
    role="heading"
  >
    19:00
  </div>
  <div
    className="event__info"
  >
    <div
      aria-level="4"
      className="event__title"
      role="heading"
    >
      testEvent
      <span
        className="event__title--free"
      />
    </div>
    <div
      className="event__info__city-name"
    >
      Mallorca - 1h
    </div>
  </div>
  <button
    className="event__sign-up btn"
    onClick={[Function]}
    type="button"
  >
    You're in
  </button>
</div>
`;
